src/metric/FIDCompoundMetric.py

```diff
@@ -105,67 +106,129 @@ def _is_ready(self, calc_mode=FID_DEFAULT_CALC_MODE) -> bool:
 
         No setup is needed for FFHQ 256/1024, if using other custom datasets `calc()`
         requires the user to run `setup()` first.
 
         Args:
+            filter_bit (int, optional): Filter bit used to select a subset of the
+                population. Filter bit is defined by the order in FilterRegistry. For example,
+                the first filter corresponds to filter bit 1. Defaults to 1 (IdentityFilter)
             calc_mode (str, optional): Either "clean", "legacy_tensorflow, or "legacy_pytorch".
                 This decides how the FID score should be calculated, i.e., using clean-fid,
                 regular tensorflow implementation, or pytorch implementation. Default is "clean" (clean-fid).
         Raises:
             ValueError: Error when non-valid `calc_mode`, valid modes are defined by `FID_CALC_MODES`.
             ValueError: Error when the name of the dataset in conjunction with the
                 specified `calc_mode` don't have a pre-computed statistic.
-
+            RuntimeError: When clean-fid library gets an error.
+            FileNotFoundError: When incorrect path is provided when moving files.
         Returns:
             Any: The FID value.
         """
         # Fetch parameters
         calc_mode = self._check_calc_mode(parameters)
 
+        # Move files to temp folder
+        uris = self._population.get_filtered_data(filter_bit)[
+            self._population.COLUMN_URI
+        ]
+        uris = [Path(str_path) for str_path in list(uris)]
+        pop_path = self._move_filtered_files(uris)
+
         # Get variables for use in FID
         ds = self.get_dataset()
-        pop_path = str(
-            Population.POPULATION_ROOT_DIR / self.get_population().get_name()
-        )
         resolution = ds.get_resolution()
         fid_score = None
 
         if type(ds).get_resolution_invariant_name() == "FFHQ" and (
             ds.get_resolution() == 256 or ds.get_resolution() == 1024
         ):
             # Use pre-computed statistic by clean-fid
-            fid_score = fid.compute_fid(
-                pop_path,
-                dataset_name=type(ds).get_resolution_invariant_name(),
-                dataset_res=resolution,
-                mode=calc_mode,
-                dataset_split="trainval70k",
-                num_workers=NUM_WORKERS,
-            )
+            try:
+                fid_score = fid.compute_fid(
+                    str(pop_path),
+                    dataset_name=type(ds).get_resolution_invariant_name(),
+                    dataset_res=resolution,
+                    mode=calc_mode,
+                    dataset_split="trainval70k",
+                    num_workers=NUM_WORKERS,
+                )
+            except:
+                self._move_filtered_files_back(pop_path)
+                raise RuntimeError(
+                    "Something went wrong when calculating FID using clean-fid library."
```

Comment:
you can still print this string to error-stream before, i.e., `sys.stderr.write("MESSAGE\n")` <- NOTE NEWLINE

Reply:
just print